source/main.cpp

```diff
@@ -120,6 +120,13 @@ int main(int argc, char *argv[])
 	Files::Init(argv);
 
 	try {
+		// Initialize the necessary threading, which also needs to be
+		// correctly destroyed after use.
+		TaskQueue::Init();
+		struct TaskQueueDeleter {
+			~TaskQueueDeleter() { TaskQueue::Destroy(); }
+		} _;
```

Comment:
Why not a destructor in the TaskQueue class/object itself?
And why not perform init in the constructor of the TaskQueue?